### PR TITLE
kube-proxy disclaimer about cleanup

### PR DESCRIPTION
**Pull request by jayunit100**
Add disclaimer about kube-proxy 
fixes https://github.com/kubernetes/kubernetes/issues/102314

### CODE REVIEW

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-28147--kubernetes-io-master-staging.netlify.app/docs/concepts/services-networking/service/#why-not-use-round-robin-dns

File: content/en/docs/concepts/services-networking/service.md [hunk 1]
```diff
@@ -241,9 +241,18 @@ There are a few reasons for using proxying for Services:
   on the DNS records could impose a high load on DNS that then becomes
   difficult to manage.
 
+Below we'll describe how various kube-proxy implementations work.  In general
+you should note that, when running `kube-proxy`, kernel level rules may be
+modified (for example, iptables rules might get created), which won't get cleaned up,
```
Comment (kbhawkey) [inline content/en/docs/concepts/services-networking/service.md:246]: Hi @jayunit100 . Some feedback:
What do you think about removing the first sentence and possibly place this paragraph (or part of the paragraph) in a Note shortcode to callout the content?

Line 247, extra spaces before "Thus"
Line 249, fix typo, "low level", privileged network ...
Line 249, extra spaces before "Although"

File: content/en/docs/concepts/services-networking/service.md [hunk 2]
```diff
@@ -241,9 +241,18 @@ There are a few reasons for using proxying for Services:
   on the DNS records could impose a high load on DNS that then becomes
   difficult to manage.
 
+Below we'll describe how various kube-proxy implementations work.  In general
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md:244]: nit ([avoid using “we”](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we)):
```suggestion
Later in this page you can read about various kube-proxy implementations work. Overall,
```

File: content/en/docs/concepts/services-networking/service.md [hunk 3]
```diff
@@ -241,9 +241,18 @@ There are a few reasons for using proxying for Services:
...
+you should note that, when running `kube-proxy`, kernel level rules may be
+modified (for example, iptables rules might get created), which won't get cleaned up, 
+in some cases until you reboot.  Thus, running kube-proxy is something that should
+only be done by an administrator which understands the consequences of having a
+low leve, priveliged network proxying service on a machine.  Although the `kube-proxy`
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md:249]:
```suggestion
low level, privileged network proxying service on a computer.  Although the `kube-proxy`
```

Comment (sftim) [reply content/en/docs/concepts/services-networking/service.md:249]: Maybe don't mention the unofficial `cleanup` subcommand at all (if it's not supported).

File: content/en/docs/concepts/services-networking/service.md [hunk 4]
```diff
@@ -241,9 +241,18 @@ There are a few reasons for using proxying for Services:
...
+executable supports a `cleanup` function, this function is not an official feature and
+thus is only available to use as-is.
+
 ### User space proxy mode {#proxy-mode-userspace}
 
-In this mode, kube-proxy watches the Kubernetes control plane for the addition and
+In this (legacy) mode, kube-proxy watches the Kubernetes control plane for the addition and
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md:255]: Is this mode legacy, and is that true on all platforms (Windows etc)?

**Review by sftim — COMMENTED** [review]
@jayunit100 would you be willing to incorporate the feedback so far?

File: content/en/docs/concepts/services-networking/service.md (hunk 4 above)
Comment (jayunit100) [reply content/en/docs/concepts/services-networking/service.md:255]: yeah... its definetly legacy for all

File: content/en/docs/concepts/services-networking/service.md (hunk 1 above)
Comment (jayunit100) [reply content/en/docs/concepts/services-networking/service.md:246]: sure, commit a suggestion?

**Comment by jayunit100** [comment]
can we merge this as-is ? feel like other improvements can happen in another thread

**Review by thockin — COMMENTED** [review]
LGTM for message

**Comment by kbhawkey** [comment]
Thanks @jayunit100 .
Lets move this PR forward and address the nits in a follow-up pull request.
/lgtm

**Pull request merged**